lib/contact.dart

```diff
@@ -220,6 +232,52 @@ class Contact {
         'groups': groups.map((x) => x.toJson()).toList(),
       });
 
+  Contact copyWith({
+    String? id,
+    String? displayName,
+    Uint8List? thumbnail,
+    Uint8List? photo,
+    bool? isStarred,
+    Name? name,
+    List<Phone>? phones,
+    List<Email>? emails,
+    List<Address>? addresses,
+    List<Organization>? organizations,
+    List<Website>? websites,
+    List<SocialMedia>? socialMedias,
+    List<Event>? events,
+    List<Note>? notes,
+    List<Account>? accounts,
+    List<Group>? groups,
+    bool? thumbnailFetched,
+    bool? photoFetched,
+    bool? isUnified,
+    bool? propertiesFetched,
+  }) {
+    return Contact(
+      id: id ?? this.id,
+      displayName: displayName ?? this.displayName,
+      thumbnail: thumbnail ?? this.thumbnail,
+      photo: photo ?? this.photo,
```

Comment:
this does not work if you want to set the photo or thumbnail to `null`.
```dart
final copy = contact.copyWith(photo: null);
```
If the original photo wasn't `null`, this implementation will not set it to `null`.

For this to work, you need to some kind of wrapper around the actual value. 
You can use a Function for that. Something like this:
```dart
  Contact copyWith({
    Uint8List? Function()? photo,
  }) {
    return Contact(
      photo: photo != null ? photo() : this.photo,
    );
  }
```

Reply:
Yes well spotted @jjochen , i think following Remi's answer [here](https://stackoverflow.com/questions/63928513/differentiate-undefined-and-null-in-dart) which the package "freezed" uses under the hood would be best: , just commited that with tests.

Though it might be a good idea to make all classes use freezed instead for a more robust management, and override any methods when needed, it's quite a bit of a big change that i am unable to take on at the moment, so perhaps that would be be on the plan for future development.